### PR TITLE
KRACOEUS-8802: refresh dialog and reset sub award period details on delete and replace

### PR DESCRIPTION
**Pull request by rmancher**
KRACOEUS-8802
Subaward - refresh dialog on delete and replace.
Also reset sub award period details on delete and replace.

### CODE REVIEW

File: coeus-impl/src/main/java/org/kuali/coeus/propdev/impl/budget/subaward/ProposalBudgetSubAwardController.java [hunk 1]
```diff
@@ -122,6 +122,7 @@ public ModelAndView syncFromPdf(@RequestParam("subAwardNumber") Integer subAward
 	public ModelAndView removeAttachment(@RequestParam("subAwardNumber") Integer subAwardNumber, @ModelAttribute("KualiForm") ProposalBudgetForm form) {
 		BudgetSubAwards subAward = getSubAwardByNumber(subAwardNumber, form);
 		getPropDevBudgetSubAwardService().removeSubAwardAttachment(subAward);
+	    form.setDialogDataObject(null);
```
Comment (jwillia) [inline coeus-impl/src/main/java/org/kuali/coeus/propdev/impl/budget/subaward/ProposalBudgetSubAwardController.java]: should we be setting the entire dialogDataObject to null when we remove an attachment? would this potentialy remove manually entered data not associated with the attachment?

Comment (rmancher) [inline coeus-impl/src/main/java/org/kuali/coeus/propdev/impl/budget/subaward/ProposalBudgetSubAwardController.java]: I did not see any issues, but looking at it I think it is better to set the updated version.
Will make changes

**Comment by rmancher** [comment]
Addressed comments

**Comment by blackcathacker** [comment]
@rmancher I see you add a description to the commit on separate lines, but if you could do a short desc on the first line with the jira # it would make the commit and the PR much easier to read.

File: coeus-impl/src/main/java/org/kuali/coeus/propdev/impl/budget/subaward/PropDevPropDevBudgetSubAwardServiceImpl.java [hunk 2]
```diff
@@ -419,6 +429,9 @@ protected Map extractAttachments(PdfReader reader)throws IOException{
     @Override
     public boolean updateSubAwardBudgetDetails(Budget budget, BudgetSubAwards budgetSubAward, List<String[]> errors) throws Exception {
         boolean result = true;
+        //reset current line items if replacing with a new one.
+        resetSubAwardPeriodDetails(budgetSubAward);
```
Comment (blackcathacker) [inline coeus-impl/src/main/java/org/kuali/coeus/propdev/impl/budget/subaward/PropDevPropDevBudgetSubAwardServiceImpl.java]: Might we want to do this after we are sure the attachment can at least be parsed?

**Comment by rmancher** [comment]
Addressed comments

**Pull request merged**